core/synchronizers.py

```diff
@@ -28,16 +44,121 @@ def test_connection(self) -> None:
         """
         pass
 
+    @abstractmethod
+    def get_external_user_info(self, oidc_id: str) -> Dict[str, str]:
+        """
+        Should return a dictionary with the user information
+        """
+        pass
+
+
+def update_user_or_contact(
```

Comment:
It feels strange to have methods called `update_user_or_contact` both inside and outside the AccountSynchronizer.

Is there a reason why these had to be defined outside the class?